components/clarinet-deployments/src/onchain/mod.rs

```diff
@@ -840,68 +835,90 @@ pub fn apply_on_chain_deployment(
                 }
             };
         }
+        let mut last_stacks_chain_check_at_height = 0;
+        let mut last_bitcoin_chain_check_at_height = 0;
 
         loop {
-            let (burn_block_height, stacks_tip_height) = match stacks_rpc.get_info() {
+            let (bitcoin_tip_height, stacks_tip_height) = match stacks_rpc.get_info() {
                 Ok(info) => (info.burn_block_height, info.stacks_tip_height),
                 _ => {
                     std::thread::sleep(std::time::Duration::from_secs(delay_between_checks));
                     continue;
                 }
             };
 
-            // If no block has been mined since `delay_between_checks`,
-            // avoid flooding the stacks-node with status update requests.
-            if burn_block_height <= current_bitcoin_block_height {
-                std::thread::sleep(std::time::Duration::from_secs(delay_between_checks));
-                continue;
-            }
-
-            current_bitcoin_block_height = burn_block_height;
-            current_block_height = stacks_tip_height;
-
             let mut keep_looping = false;
 
-            for (_txid, tracker) in ongoing_batch.iter_mut() {
-                match &tracker.status {
-                    TransactionStatus::Broadcasted(TransactionCheck::ContractPublish(
-                        deployer,
-                        contract_name,
-                    )) => {
-                        let deployer_address = deployer.to_address();
-                        let res = stacks_rpc.get_contract_source(&deployer_address, contract_name);
-                        match res {
-                            Ok(_contract) => {
-                                tracker.status = TransactionStatus::Confirmed;
-                                let _ = deployment_event_tx
-                                    .send(DeploymentEvent::TransactionUpdate(tracker.clone()));
+            // Handle Stacks releated checks
+            if stacks_tip_height > last_stacks_chain_check_at_height {
+                for (_, tracker) in ongoing_batch.iter_mut() {
+                    let TransactionStatus::Broadcasted(brodcasting_status, _) = &tracker.status
+                    else {
+                        continue;
+                    };
```

Comment:
Is this a normal syntax?

Reply:
Yep, we don't want to execute the subsequent line if the chain height is still the same